requirements.txt

```
@@ -11,19 +11,19 @@ chardet==5.2.0
 # Documentation
 mkdocs==1.6.1
 mkautodoc==0.2.0
-mkdocs-material==9.5.47
+mkdocs-material==9.6.18
 
 # Packaging
-build==1.2.2.post1
-twine==6.0.1
+build==1.3.0
+twine==6.1.0
 
 # Tests & Linting
-coverage[toml]==7.6.1
-cryptography==44.0.1
-mypy==1.13.0
-pytest==8.3.4
-ruff==0.8.1
-trio==0.27.0
+coverage[toml]==7.10.6
+cryptography==45.0.7
+mypy==1.17.1
+pytest==8.4.1
+ruff==0.12.11
```

This upgrade is failing the GitHub Actions tests.
```suggestion
ruff==0.8.1
```
Or merge #3651.
* #3651